R/heatmap.R

```diff
@@ -0,0 +1,291 @@
+plot.ccf.hm <- function(
+    CCF.df,
+    CCF.threshold = NULL,
+    cluster.dim = 'both',
+    cluster.method = 'complete',
+    dist.method = 'euclidean',
+    colour.scheme = NULL,
+    xaxis.lab = NULL,
+    xlab.label = 'Mutations',
+    ...
+    ) {
+
+    if (!is.null(CCF.threshold)) {
+        CCF.df[CCF.df <= CCF.threshold] <- 0;
+        }
+    col.labels <- seq(0, 1, .2);
+    sample.names <- colnames(CCF.df);
+
+    heatmap.colours <- if (!is.null(colour.scheme)) {
+        colour.scheme;
+    } else {
+        default.heatmap.colours();
+        }
+
+    hm <- BoutrosLab.plotting.general::create.heatmap(
+        filename = NULL,
+        x = CCF.df,
+        force.clustering = TRUE,
+        cluster.dimensions = cluster.dim,
+        clustering.method = cluster.method,
+        rows.distance.method = dist.method,
+        cols.distance.method = dist.method,
+        xaxis.lab = xaxis.lab,
+        xlab.label = xlab.label,
+        xlab.cex = 1,
+        xaxis.cex = 0.6,
+        xaxis.fontface = 1,
+        xaxis.rot = 90,
+        yaxis.lab = sample.names,
+        ylab.cex = 1,
+        yaxis.cex = 0.6,
+        yaxis.fontface = 1,
+        colourkey.cex = 0.6,
+        colour.scheme = heatmap.colours,
+        left.padding = 1,
+        right.padding = 1,
+        resolution = 3000,
+        width = 9,
```

Comment:
We want to make the `height` and `width` customizable since the H:W ratio depends on the number of samples and mutations